### DISCUSSION TITLE
Best DB strategy for insights/panels operations with event rows

### DISCUSSION THREAD

**Discussion by mmmaxxxe** [directus/directus#12383] [Archive]
Hi, 
i'm quite new with directus. Love the crud application and API.
I'm planning to migrate an existing database to Directus. (postgres / some millions of records)  The problem is that the biggest part of the db stores event rows, as it registers mainly access/visits and sort of analytics in the following matter:

id_event | event_datetime | user_generated | event_type

I woud like to display some data in a dashboard/panel. After seeing examples on the demo instance, as far as i can see, there is no query builder in directus so cannot implement a consolidated table / view where i can count/sum those events in a date interval, for example

time_interval | number_of_events | user_generated | event_type

What is the best strategy to accomplish this? 
- View in the db? (does directus wraps everything on his own if i create a view externally from directus?)
- cron job populating a separate table?
- other?

Thanks in advance for the attention, hope that someone can address me on the best solution.

**Comment by rijkvanzanten** [comment]
Directus doesn't currently support creating/reading from Views, so your best option would be option 2, where you'd effectively run some automation to create the "report" rows in a secondary table, and display those in dashboards 🙂

**Reply by mmmaxxxe** [reply]
Hi Rijk thanks for your reply and for the support. Very appreciated.
